### PR TITLE
Fix release URL links

### PR DESCRIPTION
**Pull request by gregone**
Fixes https://github.com/hashicorp/waypoint/issues/1827

### CODE REVIEW

File: ui/app/components/app-card/release.hbs [hunk 1]
```diff
@@ -25,7 +25,7 @@
   <:actions>
     {{#if @model.url}}
       <ExternalLink
-        href="https://{{@model.url}}"
+        href={{@model.url}}
         title="Visit https://{{@model.url}}"
```
Comment (koesbong) [inline ui/app/components/app-card/release.hbs:29]: Should the title here be updated too without the `https://`?

**Review by jgwhite — APPROVED** [review]

**Comment by jgwhite** [comment]
I wonder if `<AppCard::Deployment>` has the same issue.

**Comment by denysvitali** [comment]
Uh! Yes it does :(

**Comment by jgwhite** [comment]
I just tested this against waypoint+k8s and it seems like `deployment.preload.deploy_url` does *not* include the protocol, while `release.url` *does*. I found I kept forgetting which way around it was. Maybe it would be worth us writing an `ensure-protocol` helper, or making `ExternalLink` automatically add the protocol only if it’s missing?

```hbs
{{ensure-protocol "http://example.com"}} {{! => http://example.com }}
{{ensure-protocol "example.com"}}        {{! => https://example.com }}
```

**Comment by denysvitali** [comment]
(I opened #1834 in the meantime)

IMHO, an object called `xyzUrl` should be an URL. That means that the protocol should be included.

I guess that we need to check and make sure that our plugins always return an URL rather than having an hackish way in the frontend to mitigate the issue.

cc/ @mitchellh, @krantzinator, @briancain  ?

**Comment by jgwhite** [comment]
Now that `deployment.url` exists (thanks @denysvitali!), we should probably transition away from `deployment.preload.deploy_url` and use `deployment.url` directly instead.

**Comment by denysvitali** [comment]
I'm not sure about that, I always though that the `deployment.preload.deploy_url` was for Horizon: but maybe we can remove this `preload` part and start using `deployment.url` directly as you mentioned, yes :)

**Comment by jgwhite** [comment]
> I'm not sure about that, I always though that the `deployment.preload.deploy_url` was for Horizon: but maybe we can remove this `preload` part and start using `deployment.url` directly as you mentioned, yes :)

I guess this is another thing we should clarify. An explicit `horizon_url` would probably be clearer.

For the purposes of this PR, it’s worth nothing that the changes #1387 weren’t back-ported, so we’ll need to stick with `deployment.preload.deploy_url` for the purposes of this PR either way.

**Comment by jgwhite** [comment]
> I guess that we need to check and make sure that our plugins always return an URL rather than having an hackish way in the frontend to mitigate the issue.

I agree but I also don’t think there’s any harm in adding the frontend mitigation too.

**Comment by denysvitali** [comment]
This will solve the issues we're having in the short term, which seems a win-win situation. We can then start to migrate everything to use proper URLs and expect that past a certain Waypoint version, only URLs are accepted on that field - so that plugins have time to adapt.

**Comment by gregone** [comment]
Are we still good to merge this one?

**Comment by jgwhite** [comment]
@gregone yeah, let’s merge this.

**Pull request merged**